shared-data/js/helpers/parseProtocolCommands.ts

```diff
@@ -367,26 +367,66 @@ export function parseLiquidsInLoadOrder(
     (command): command is LoadLiquidRunTimeCommand =>
       command.commandType === 'loadLiquid'
   )
-  const loadedLiquids = liquids.map((liquid, index) => {
-    return {
-      ...liquid,
-      displayColor:
-        liquid.displayColor ??
-        DEFAULT_LIQUID_COLORS[index % DEFAULT_LIQUID_COLORS.length],
+
+  // Assign default colors to liquids
+  const loadedLiquids = liquids.map((liquid, index) => ({
+    ...liquid,
+    displayColor:
+      liquid.displayColor ??
+      DEFAULT_LIQUID_COLORS[index % DEFAULT_LIQUID_COLORS.length],
+  }))
+  const acc: ParsedLiquid[] = []
+  const seenLiquids = new Set<string>()
+  const wellToLiquids: Record<string, Record<string, Set<string>>> = {}
+
+  loadLiquidCommands.forEach(cmd => {
+    const liquidId = cmd.params.liquidId
+
+    // Add base liquids in load order
+    if (!seenLiquids.has(liquidId)) {
+      seenLiquids.add(liquidId)
+      const liquid = loadedLiquids.find(l => l.id === liquidId)
+      if (liquid) acc.push(liquid)
+    }
+
+    // Track well-to-liquid mapping for mixed liquids
+    const labwareId = cmd.params.labwareId
+    if (!wellToLiquids[labwareId]) wellToLiquids[labwareId] = {}
+
+    for (const well of Object.keys(cmd.params.volumeByWell)) {
+      if (!wellToLiquids[labwareId][well])
+        wellToLiquids[labwareId][well] = new Set()
+      wellToLiquids[labwareId][well].add(liquidId)
     }
   })
 
-  return reduce<LoadLiquidRunTimeCommand, ParsedLiquid[]>(
-    loadLiquidCommands,
-    (acc, command) => {
-      const liquid = loadedLiquids.find(
-        liquid => liquid.id === command.params.liquidId
-      )
-      if (liquid != null && !acc.some(item => item === liquid)) acc.push(liquid)
-      return acc
-    },
-    []
-  )
+  // Map to track unique mixed combinations
+  const mixedMap = new Map<string, ParsedLiquid>()
+
+  Object.values(wellToLiquids).forEach(wellMap => {
+    Object.values(wellMap).forEach(liquidIds => {
+      if (liquidIds.size > 1) {
+        const sortedIds = Array.from(liquidIds).sort()
+        const key = sortedIds.join('-')
+
+        if (!mixedMap.has(key)) {
+          const liquidNames = sortedIds.map(
+            id => loadedLiquids.find(l => l.id === id)?.displayName || id
+          )
+
+          const mixedId = `mixed-${sortedIds.join('-')}`
+
+          mixedMap.set(key, {
+            id: mixedId,
+            displayColor: '#737578',
```

Comment:
```suggestion
            displayColor: MIXED_WELL_COLOR,
```
Yeah, think we should mimic PD here and pull this into a constant.